modules/core/src/main/scala/com.snowplowanalytics.snowplow.snowflake/Environment.scala

```diff
@@ -52,19 +63,47 @@ object Environment {
       badSink <- toSink(config.output.bad.sink).onError(_ => Resource.eval(appHealth.beUnhealthyForRuntimeService(RuntimeService.BadSink)))
       metrics <- Resource.eval(Metrics.build(config.monitoring.metrics))
       tableManager <- Resource.eval(TableManager.make(config.output.good, appHealth, config.retries))
-      channelOpener <- Channel.opener(config.output.good, config.batching, config.retries, appHealth)
-      channelProvider <- Channel.provider(channelOpener, config.retries, appHealth)
+      cpuParallelism    = chooseCpuParallelism(config)
+      uploadParallelism = chooseUploadParallelism(config)
```

Comment:
```suggestion
      cpuParallelism    = multiplyByCpuAndRoundUp(config.cpuParallelismFactor)
      uploadParallelism = multiplyByCpuAndRoundUp(config.batching.uploadParallelismFactor)
```
Do we really need these functions ?

Reply:
I'd like to keep them because I like the scaladoc explaining my thinking.